script/create_changelog

```diff
@@ -0,0 +1,250 @@
+}
+
+#is_tag_belongs_to_release check if tag=$2 is part of release=$1
+# if tag is v1.1.0-RC1 and release is v1.1.0 then it will return 0
+# if tag is v1.0.0-RC1 and release is v1.1.0 then it will return 1
+# if tag is v1.1.0-custom-RC1 and release is v1.1.0 then it will return 1
+is_tag_belongs_to_release() {
+	release=$1
+	tag=$2
+
+	[[ -z $release ]] && echo 1 && return
+	[[ -z $tag ]] && echo 1 && return
+
+	[[ "$(echo $tag | cut -d '-' -f1)" == $release ]] && echo 0 || echo 1
+}
+
+#add_changelog add the changelog from files in folder=$1 to file=$2
+add_changelog() {
+	cl_dir=$1
+	cl_file=$2
+
+	[[ -d $cl_dir ]] || return
+	[[ -f $cl_file ]] || return 
+
+	echo "Adding changelog from $cl_dir"
+
+	for i in `find $cl_dir -type f`; do
+		pr_number=$(echo $i | cut -d  '/' -f4 |awk -F '-' '{print $1}')
+		usr_name=$(echo $i | cut -d  '/' -f4 |awk -F '-' '{print $2}')
+		msg=$(cat $i)
+
+		url_usr=https://github.com/$usr_name
+		url_pr=https://github.com/$GIT_REPO/pull/$pr_number
+		echo "* $msg ([#$pr_number]($url_pr),[@$usr_name]($url_usr))" >> $cl_file
+		echo -n "- $url_pr\r\n" >>  $pr_msg_file
+	done
+}
+
+#create_changelog_commit update the changelog and create a commit for the same
+create_changelog_commit() {
+	echo "Generating Changelog for $TAG"
+	local dest=$RELEASED_CHANGELOG_DIR/$TAG
+
+	[[ -d $CHANGELOG_DIR ]] || return 3
+	[[ -d $UNRELEASED_CHANGELOG_DIR ]] || return 3
+	[[ -d $RELEASED_CHANGELOG_DIR ]] || mkdir $RELEASED_CHANGELOG_DIR
+	[[ -d $dest ]] || mkdir $dest
+
+	ls $UNRELEASED_CHANGELOG_DIR/* > /dev/null 2>&1 && cp $UNRELEASED_CHANGELOG_DIR/* $dest
+
+	cl=$(mktemp)
+	echo "$TAG / $(date +'%Y-%m-%d')" >> $cl
+	echo "========================" >> $cl
+
+	add_changelog $dest $cl
+
+	if [ $(is_stable_tag $TAG) -eq 0 ]; then
+		# We are creating a changelog for stable release
+		# so we also need to append changelog of RC tag
+		for i in `find $RELEASED_CHANGELOG_DIR  -maxdepth 1 -mindepth 1 -type d`; do
+			local ltag=$(echo $i | cut -d '/' -f3)
+			[[ $(is_rc_tag $ltag) -ne 0 ]] && continue
+			[[ $(is_tag_belongs_to_release $TAG $ltag) -ne 0 ]] && continue
+			add_changelog $i $cl
+		done
+	fi
+	temp_file=$(mktemp)
+	[[ -f $ROOT_CHANGELOG ]] && echo -e "\n\n" >> $cl || touch $ROOT_CHANGELOG
+	cat <(cat $cl) $ROOT_CHANGELOG  > $temp_file && mv $temp_file $ROOT_CHANGELOG
+	ls $UNRELEASED_CHANGELOG_DIR/* > /dev/null  2>&1 && rm $UNRELEASED_CHANGELOG_DIR/*
+	git add $CHANGELOG_DIR
+	git add $ROOT_CHANGELOG
+	git commit -s -m "Updating changelog for $TAG"
+	rm -rf ${cl} ${temp_file} > /dev/null 2>&1
+}
+
+# update_branch adds new remote and create/checkout branch=$BRANCH
+update_branch() {
+	echo "Creating branch $BRANCH"
+	git config --global user.name ${USERNAME}
+	git config --global user.email ${EMAIL}
+	local repo_name=$(echo $GIT_REPO | cut -d '/' -f2)
+	git remote add upstream https://${USERNAME}:${TOKEN}@github.com/$USERNAME/$repo_name.git > /dev/null 2>&1
+	git branch $BRANCH && git checkout $BRANCH
```

Comment:
If we are on the release branch, suppose `v1.10.x`, then `git branch` will create a new branch from `v1.10.x` right? not master.

Reply:
yes.

Reply:
cherry_pick will add commit to production branch and delete old branch an create a new branch=$BRANCH from production branch
This is required because release branch and production branch may have different commits. Ideally master branch will have all the changelog, but release branch will have subset of that.

Reply:
@akhilerm I've updated the script to raise a PR against release branch also.